src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Pass.cpp

```diff
@@ -15,23 +15,70 @@ namespace orbit
 {
 
 Pass::Pass(
-    const Pass::Type& aType,
     const Integer& aRevolutionNumber,
-    const Interval& anInterval,
-    const Instant& anInstantAtDescendingNode,
+    const Instant& anInstantAtAscendingNode,
     const Instant& anInstantAtNorthPoint,
-    const Instant& anInstantAtSouthPoint
+    const Instant& anInstantAtDescendingNode,
+    const Instant& anInstantAtSouthPoint,
+    const Instant& anInstantAtPassBreak
 )
-    : type_(aType),
-      revolutionNumber_(aRevolutionNumber),
-      interval_(anInterval),
-      instantAtDescendingNode_(anInstantAtDescendingNode),
+    : revolutionNumber_(aRevolutionNumber),
+      instantAtAscendingNode_(anInstantAtAscendingNode),
       instantAtNorthPoint_(anInstantAtNorthPoint),
-      instantAtSouthPoint_(anInstantAtSouthPoint)
+      instantAtDescendingNode_(anInstantAtDescendingNode),
+      instantAtSouthPoint_(anInstantAtSouthPoint),
+      instantAtPassBreak_(anInstantAtPassBreak)
 {
-    if ((aType == Pass::Type::Complete) && (!anInstantAtNorthPoint.isDefined() || !anInstantAtSouthPoint.isDefined()))
+    if (!anInstantAtNorthPoint.isDefined() && !anInstantAtSouthPoint.isDefined() &&
+        !anInstantAtDescendingNode.isDefined() && !anInstantAtAscendingNode.isDefined() &&
+        !anInstantAtPassBreak.isDefined() && !revolutionNumber_.isDefined())
+    {
+        type_ = Pass::Type::Undefined;
+    }
+    else if (!anInstantAtNorthPoint.isDefined() || !anInstantAtSouthPoint.isDefined() ||
+        !anInstantAtDescendingNode.isDefined() || !anInstantAtAscendingNode.isDefined() ||
+        !anInstantAtPassBreak.isDefined())
+    {
```

Comment:
```suggestion
    else if ((!anInstantAtNorthPoint.isDefined() || !anInstantAtSouthPoint.isDefined() ||
        !anInstantAtDescendingNode.isDefined() || !anInstantAtAscendingNode.isDefined() ||
        !anInstantAtPassBreak.isDefined()) && revolutionNumber_.isDefined())
    {
```

Because revolution number always has to be defined right?

Otherwise i think this allows revolution number to be undefined and have a pass with a `Partial` type

Reply:
Yeah good catch!